cmd/kube-proxy/app/server.go

```diff
@@ -344,6 +349,12 @@ with the apiserver API to configure the proxy.`,
 			verflag.PrintAndExitIfRequested()
 			utilflag.PrintFlags(cmd.Flags())
 
+			// This can happen only on Windows if it cannot register or unregister
+			// as a Windows service. On Linux err will always be nil.
+			if err := initForOS(); err != nil {
```

Comment:
Is a constructor really the best place to call this? I'd expect to see init calls like this in a bootstrapping function, like Run. @ncdc

Reply:
@mtaufen this is part of `Run`. When I first looked at your comment, it looked like GH was showing it as part of `NewOptions` (so my reaction was in line with your comment), but if you expand the lines above this diff, you'll see it's in `Run`.... right?

Reply:
Oh GitHub xD. You're right.